core/ide/che-core-ide-api/src/main/java/org/eclipse/che/ide/api/app/AppContext.java

```diff
@@ -21,28 +28,114 @@
  *
  * @author Vitaly Parfonov
  * @author Artem Zatsarynnyi
+ * @author Vlad Zhukovskyi
  */
 public interface AppContext {
 
-    /** Returns list of start-up actions with parameters that comes form URL during IDE initialization. */
-    List<StartUpAction> getStartAppActions();
+    /**
+     * Returns the workspace root container, which is holder of registered projects.
+     *
+     * @return the workspace root
+     * @since 4.4.0
+     */
+    @Beta
+    Container getWorkspaceRoot();
 
-    WorkspaceDto getWorkspace();
+    /**
+     * Returns the registered projects in current workspace. If no projects were registered before,
+     * then empty array is returned.
+     *
+     * @return the registered projects
+     * @see Container#newProject()
+     * @since 4.4.0
+     */
+    @Beta
+    Project[] getProjects();
 
-    void setWorkspace(WorkspaceDto workspace);
+    /**
+     * Returns the resource which is in current context. By current context means, that resource may be
+     * in use in specified part if IDE. For example, project part may provide resource which is under
+     * selection at this moment, editor may provide resource which is open, full text search may provide
+     * resource which is under selection.
+     * <p/>
+     * If specified part provides more than one resource, then last selected resource is returned.
+     * <p/>
+     * May return {@code null} if there is no resource in context.
+     *
+     * @return the resource in context
+     * @see Resource
+     * @see #getResources()
+     * @since 4.4.0
+     */
+    @Beta
+    Resource getResource();
 
-    /** Returns id of current workspace of throws IllegalArgumentException if workspace is null. */
-    String getWorkspaceId();
+    /**
+     * Returns the resources which are in current context. By current context means, that resources may be
+     * in use in specified part if IDE. For example, project part may provide resources which are under
+     * selection at this moment, editor may provide resource which is open, full text search may provide
+     * resources which are under selection.
+     * <p/>
+     * If specified part provides more than one resource, then all selected resources are returned.
```

Comment:
I don't think we need to specify if we have more than one resource as this method is for "all resources"